src/Symfony/Component/HttpKernel/DependencyInjection/Extension.php

```diff
@@ -17,6 +17,8 @@
  * Allow adding classes to the class cache.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @internal since Symfony 7.1, to be deprecated in 8.1; use Symfony\Component\DependencyInjection\Extension\Extension instead
```

Comment:
We cannot deprecate this class right away in 7.1 because we rely on it in FrameworkBundle and others.
Making it first internal the deprecate in 8.1 provides a smooth upgrade path.

Reply:
Actually, if it is already a no-op, FrameworkBundle could stop registering it.

Reply:
if you mean AddAnnotatedClassesToCachePass, that's done is this PR

Reply:
ah, I see. This is about the base class then.

Marking as `@internal` first which removes it from our BC policy before deprecating it only in 8.1 looks weird to me. If we don't deprecate it, we might keep it as non-internal.

Reply:
it's `@internal since`, which means its deprecated to use, but we signals to us we won't remove it in 8.0 for BC reasons

Reply:
but `@internal since 7.1` would become `@internal` in 8.0, which _can_ be removed.

Reply:
can but doesn't have to, while deprecated means it should be removed
I'm looking for a smoother plan

Reply:
fine by me